### ISSUE TITLE
Version 6.0 not logging in when using Docker

### ISSUE THREAD

**Issue by jdholtz** [jdholtz/auto-southwest-check-in#159]
### Discussed in https://github.com/jdholtz/auto-southwest-check-in/discussions/158

<div type='discussions-op-text'>

<sup>Originally posted by **ctkubik** September 23, 2023</sup>
It seems like I can use reservations, but my login information is throwing an error.

I am far from a Docker expert...I've been using it for a few versions and this is pretty much all I use it for.  What am I missing here?  I thought this version stripped the dependency for Chrome...or is there something more I need to do?  

2023-09-23 17:44:56 Encountered a Too Many Requests error while logging in. Skipping reservation retrieval
2023-09-23 17:44:56 The chromedriver version (117.0.5938.62) detected in PATH at /usr/bin/chromedriver might not be compatible with the detected chrome version (117.0.5938.92); currently, chromedriver 117.0.5938.92 is recommended for chrome 117.*, so it is advised to delete the driver in PATH and retry</div>

**Comment by jdholtz** [comment]
A feature that will hopefully fix this issue is currently being worked on by the SeleniumBase maintainer and should be fixed in a few days.

**Comment by ericlamus** [comment]
I ran into this as well.

**Comment by dmytrokoren** [comment]
@jdholtz I ran into similar issue but without using account login but with confirmation number and Fn and Ln.

Failed to check in to flight 4Gxxxx for Jane Doe. Reason: Too Many Requests 429.
Check in at this url: https://mobile.southwest.com/check-in

**Comment by jdholtz** [comment]
@dmytrokoren was this ran in the Docker image? Also, does it repeatedly fail?

**Comment by ericlamus** [comment]
To clarify, I'm running this in this in a Docker container.

**Comment by jdholtz** [comment]
> I’m not sure if it fails repeatedly I only had one reservation

To check if it fails repeatedly, you can just run the Docker container again (if the reservation is valid)

**Comment by dmytrokoren** [comment]
> > I’m not sure if it fails repeatedly I only had one reservation
> 
> To check if it fails repeatedly, you can just run the Docker container again (if the reservation is valid)

**Yes. I just checked again and it is failing with same error message:**
Failed to check in to flight 4Gxxxx for Jane Doe. Reason: Too Many Requests 429.
Check in at this url: https://mobile.southwest.com/check-in

**Comment by jdholtz** [comment]
> Yes. I just checked again and it is failing with same error message:

Thanks for trying. For now you can just keep using v5.1 until I fix the issue.

**Comment by dmytrokoren** [comment]
@jdholtz I tried v5.1 and v5.0 for account login and reservations on docker, I'm getting same error Too Many Requests 429.

**Comment by jdholtz** [comment]
Strange. I’ll try later today as I have a valid check-in

**Comment by Kaden-Warner** [comment]
I got that same error on v6.0 running outside Docker. I already checked into the flight before running the script however, not sure if that has any effect. This is using reservations details not Southwest account

**Comment by jdholtz** [comment]
@Kaden-Warner do you get the same issue with v5.1? Also, do you get a message in the console that says "The chromedriver version detected in PATH…"?

**Comment by Kaden-Warner** [comment]
@jdholtz Just tested again with v5.1, ran into same issue. I'm not seeing that message in the console when running in verbose

**Comment by jdholtz** [comment]
I am not getting any issues running Docker version v5.1 when checking in. I am only getting errors on v6.0 (it still works if I run it without Docker). I tried with Chromium and Google Chrome locally on v6.0 without issues.

Are you guys getting the 429 error right away? Or does it only happen while attempting to check in? If it is right away, you may be blocked by Southwest. Try seeing if you can connect through a web browser.

**Comment by dmytrokoren** [comment]
@jdholtz on the reservations when I start the container I get this message:

Successfully scheduled the following flights to check in for Jane Doe:
Flight from Dallas (Love Field) to New York (LaGuardia) at 2023-09-25 18:20:00 UTC

Afterwards I get this error:
Failed to check in to flight 4Gxxxx for Jane Doe. Reason: Too Many Requests 429.
Check in at this url: https://mobile.southwest.com/check-in

**Comment by Kaden-Warner** [comment]
@jdholtz I am experiencing the same problem as @dymtrokoren, the error
only comes when attenpting check it.

**Comment by jdholtz** [comment]
Has it worked on v5.0 or v5.1 for you guys in the past? That is strange why it would stop working all of a sudden

**Comment by Kaden-Warner** [comment]
Today's actually my first time trying it out 😅 Attaching logs with v6.0 (v5.2 looks pretty much identical)
[auto-southwest-check-in.log](https://github.com/jdholtz/auto-southwest-check-in/files/12720882/auto-southwest-check-in.log)

**Comment by jdholtz** [comment]
Sounds good. Thanks for attaching the logs @Kaden-Warner. 

Once the original issue is fixed, I’ll let you guys know and you can try logging in (or check in if you have a valid flight). If it still doesn’t work, a separate issue can be opened.

**Comment by coleby** [comment]
Same issue with 6.0, Downgrading to 5.1 worked perfectly!

**Comment by dmytrokoren** [comment]
> Same issue with 6.0, Downgrading to 5.1 worked perfectly!

Hey, it works for you on docker?
Was it reservations or login?

**Comment by coleby** [comment]
> > Same issue with 6.0, Downgrading to 5.1 worked perfectly!
> 
> Hey, it works for you on docker? Was it reservations or login?

Yes, Docker and using my Account Login. Had the same error as everyone else using 6.0 Docker. Removed it, and used 5.1 with the same config.

I can confirm that Scheduling works and so do Email Notifications. Will hopefully know soon if fare checking / checking-in actually works!

**Comment by jdholtz** [comment]
This issue has been fixed for me on the Docker image with the `develop` tag. Can everyone running into this issue pull the latest develop image (`docker pull jdholtz/auto-southwest-check-in:develop`) and test to verify it is working for you as well?

**Comment by coleby** [comment]
> This issue has been fixed for me on the Docker image with the `develop` tag. Can everyone running into this issue pull the latest develop image (`docker pull jdholtz/auto-southwest-check-in:develop`) and test to verify it is working for you as well?

@jdholtz develop works for me as well. :shipit:

**Comment by dmytrokoren** [comment]
Not working for me :(
I'm using docker on Synology NAS, it was working fine before in July. Now not even the :develop tag is working for me.
My logs below:

2023-09-27 05:44:58 DEBUG MainProcess[log:23]: Initialized the application       
2023-09-27 05:44:58 DEBUG MainProcess[main:112]: Auto-Southwest Check-In v6.0    
2023-09-27 05:44:58 DEBUG MainProcess[main:113]: Called with 0 arguments         
2023-09-27 05:44:58 DEBUG MainProcess[config:107]: Initializing configuration fil
e                                                                                
2023-09-27 05:44:58 DEBUG MainProcess[config:135]: Reading the configuration file
2023-09-27 05:44:58 DEBUG MainProcess[config:53]: Setting check fares to True    
2023-09-27 05:44:58 DEBUG MainProcess[config:67]: Setting notification level to <
NotificationLevel.INFO: 1>                                                       
2023-09-27 05:44:58 DEBUG MainProcess[config:80]: Using 1 notification services  
2023-09-27 05:44:58 DEBUG MainProcess[config:84]: Setting retrieval interval to 2
4 hours                                                                          
2023-09-27 05:44:58 DEBUG MainProcess[config:118]: Creating configurations for 1 
accounts                                                                         
2023-09-27 05:44:58 DEBUG MainProcess[config:125]: Creating configurations for 0 
reservations                                                                     
2023-09-27 05:44:58 DEBUG MainProcess[main:142]: Monitoring 1 accounts and 0 rese
rvations                                                                         
2023-09-27 05:44:59 DEBUG Process-1[reservation_monitor:154]: Acquiring lock...  
2023-09-27 05:44:59 DEBUG Process-1[reservation_monitor:156]: Lock acquired      
2023-09-27 05:44:59 DEBUG Process-1[reservation_monitor:178]: Retrieving reservat
ions for account                                                                 
2023-09-27 05:44:59 DEBUG Process-1[webdriver:124]: Starting webdriver for curren
t session                                                                        
2023-09-27 05:45:01 DEBUG Process-1[webdriver:135]: Loading Southwest Check-In pa
ge                                                                               
2023-09-27 05:45:12 DEBUG Process-1[webdriver:79]: Logging into account to get a 
list of reservations and valid headers                                           
2023-09-27 05:45:15 DEBUG Process-1[webdriver:164]: Waiting for headers_set to be
 set                                                                             
2023-09-27 05:45:15 DEBUG Process-1[webdriver:168]: headers_set set successfully 
2023-09-27 05:45:15 DEBUG Process-1[webdriver:164]: Waiting for login_request_id 
to be set                                                                        
2023-09-27 05:45:16 DEBUG Process-1[webdriver:156]: Login response has been recei
ved                                                                              
2023-09-27 05:45:16 DEBUG Process-1[webdriver:168]: login_request_id set successf
ully                                                                             
2023-09-27 05:45:17 DEBUG Process-1[webdriver:179]: Logging in failed for an unkn
own reason                                                                       
2023-09-27 05:45:17 WARNING Process-1[reservation_monitor:187]: Encountered a Too
 Many Requests error while logging in. Skipping reservation retrieval            
2023-09-27 05:45:17 DEBUG Process-1[reservation_monitor:167]: Lock released      
2023-09-27 05:45:17 DEBUG Process-1[reservation_monitor:116]: Sleeping for 86382 
seconds

**Comment by jdholtz** [comment]
@dmytrokoren is v5.1 working for you? If not, this issue is unfortunately not the same issue you are running into because this is specifically for v6.0

**Comment by jdholtz** [comment]
@dmytrokoren could you create a separate discussion/issue for this?

**Comment by dmytrokoren** [comment]
> @dmytrokoren could you create a separate discussion/issue for this?

Done. ✅

**Comment by jdholtz** [comment]
Fixed in v6.1. Thanks to everyone who tested!
